datadog/test_utils.go

```diff
@@ -0,0 +1,26 @@
+package datadog
+
+import (
+	"fmt"
+	"time"
+)
+
+// Retry calls the call function for count times every interval while it returns true
+// Call function should return (false, nil) to indicate a success
+// (true, err) to indicate there was an error but we should retry
+// and (false, err) to indicate there was a fatal error
+func Retry(interval time.Duration, count int, call func() (bool, error)) error {
+	for i := 0; i < count; i++ {
+		shouldRetry, err := call()
```

Comment:
WDYT of using a special error type like `RetriableError? I think we should retry in all other cases anyway, and it would simplify signature changes.

Reply:
Yeah I think that makes sense. Will look into implementing that